### PR TITLE
RSCBC-25, RSCBC-42, RSCBC-24: Implement frontend

### PR DESCRIPTION
**Pull request by chvck**

### CODE REVIEW

File: sdk/couchbase/src/clients/cluster_client.rs [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+                .await?,
+            )
+        };
+
+        Ok(ClusterClient { backend })
+    }
+
+    pub async fn bucket_client(&self, name: String) -> error::Result<BucketClient> {
+        match &self.backend {
+            ClusterClientBackend::CouchbaseClusterBackend(ref backend) => {
```
Comment (Westwooo) [inline sdk/couchbase/src/clients/cluster_client.rs:52]: For my own understanding, why does this need to be `ref backend`?

Comment (chvck) [reply sdk/couchbase/src/clients/cluster_client.rs:52]: Because I messed up

File: sdk/couchbase/src/clients/cluster_client.rs [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+                "kv_connect_timeout" => {
+                    opts.connect_timeout =
+                        Some(Duration::from_millis(v[0].parse().map_err(|e| {
+                            error::Error {
+                                msg: format!("{}", e),
+                            }
+                        })?))
+                }
+                "placeholder" => {}
+                _ => (),
```
Comment (Westwooo) [inline sdk/couchbase/src/clients/cluster_client.rs:147]: Is it worth logging if extra_opts are being ignored?

File: sdk/couchbase/src/clients/collection_client.rs [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+                self.name().to_string(),
+            ),
+        ))
+    }
+}
+
+#[derive(Clone)]
+pub(crate) struct Couchbase2CollectionClient {
+    // bucket_name: String,
+    // scope_name: String,
```
Comment (Westwooo) [inline sdk/couchbase/src/clients/collection_client.rs:82]: Why doesn't this one need bucket/scope name?

File: sdk/couchbase/src/options/cluster_options.rs [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+use couchbase_core::ondemand_agentmanager::OnDemandAgentManagerOptions;
+
+#[derive(Clone, Debug, TypedBuilder)]
+#[builder(field_defaults(default, setter(into)))]
+#[non_exhaustive]
+pub struct ClusterOptions {
+    // authenticator specifies the authenticator to use with the cluster.
+    #[builder(!default)]
+    pub authenticator: Authenticator,
+    // timeout_options specifies various operation timeouts.∂
```
Comment (Westwooo) [inline sdk/couchbase/src/options/cluster_options.rs:18]: Mysterious symbol after the full stop?

**Review by Westwooo — APPROVED** [review]
Looks like it's taking shape nicely

File: sdk/couchbase/src/clients/cluster_client.rs (hunk 2 above)
Comment (chvck) [reply sdk/couchbase/src/clients/cluster_client.rs:147]: Maybe, but I've logged an issue to expand our parsing - for now this is POC.

File: sdk/couchbase/src/clients/collection_client.rs (hunk 3 above)
Comment (chvck) [reply sdk/couchbase/src/clients/collection_client.rs:82]: It does, it's effectively just stubbed out for now as POC.

**Review by Matt-Woz — APPROVED** [review]

**Pull request merged**